contracts/finance/andromeda-weighted-distribution-splitter/src/contract.rs

```diff
@@ -0,0 +1,1145 @@
+            AddressWeight {
+                recipient: Recipient::from_string(String::from("addr3")),
+                weight: Uint128::new(50),
+            }
+        );
+        assert_eq!(splitter.recipients.len(), 2);
+    }
+
+    #[test]
+    fn test_update_recipient_weight() {
+        let mut deps = mock_dependencies();
+        let env = mock_env();
+        let owner = "creator";
+
+        let recipient = vec![
+            AddressWeight {
+                recipient: Recipient::from_string(String::from("addr1")),
+                weight: Uint128::new(40),
+            },
+            AddressWeight {
+                recipient: Recipient::from_string(String::from("addr2")),
+                weight: Uint128::new(60),
+            },
+            AddressWeight {
+                recipient: Recipient::from_string(String::from("addr3")),
+                weight: Uint128::new(50),
+            },
+        ];
+        let msg = ExecuteMsg::UpdateRecipients {
+            recipients: recipient.clone(),
+        };
+
+        let deps_mut = deps.as_mut();
+        ADOContract::default()
+            .instantiate(
+                deps_mut.storage,
+                deps_mut.api,
+                mock_info(owner, &[]),
+                BaseInstantiateMsg {
+                    ado_type: "splitter".to_string(),
+                    operators: None,
+                    modules: None,
+                    primitive_contract: None,
+                },
+            )
+            .unwrap();
+
+        let info = mock_info("incorrect_owner", &[]);
+        let res = execute(deps.as_mut(), env.clone(), info, msg);
+        assert_eq!(ContractError::Unauthorized {}, res.unwrap_err());
+
+        let info = mock_info(owner, &[]);
+
+        let msg = ExecuteMsg::UpdateRecipients {
+            recipients: recipient.clone(),
+        };
+        let splitter = Splitter {
+            recipients: recipient.clone(),
+            locked: false,
+        };
+
+        SPLITTER.save(deps.as_mut().storage, &splitter).unwrap();
+
+        let _res = execute(deps.as_mut(), env.clone(), info.clone(), msg).unwrap();
+
+        // User not found
+        let msg = ExecuteMsg::UpdateRecipientWeight {
+            recipient: AddressWeight {
+                recipient: Recipient::from_string(String::from("addr4")),
+                weight: Uint128::new(100),
+            },
+        };
+        let err = execute(deps.as_mut(), env.clone(), info.clone(), msg).unwrap_err();
+        assert_eq!(err, ContractError::UserNotFound {});
+
+        // Invalid weight
+        let msg = ExecuteMsg::UpdateRecipientWeight {
+            recipient: AddressWeight {
+                recipient: Recipient::from_string(String::from("addr1")),
+                weight: Uint128::zero(),
+            },
+        };
+        let err = execute(deps.as_mut(), env.clone(), info.clone(), msg).unwrap_err();
+        assert_eq!(err, ContractError::InvalidWeight {});
+
+        // Locked contract
```

Comment:
Same as above. Just so we have better traceability on exactly what is erroring in our CI.